### PR TITLE
feat(analytics): add google analytics

### PR DESCRIPTION
**Pull request by gozineb**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Review by StanGirard — DISMISSED** [review]
LGTM

**Comment by StanGirard** [comment]
@gozineb don't you need to add Google analytics to SCP ?

File: frontend/app/user/components/ApiKeyConfig/hooks/useApiKeyConfig.ts [hunk 1]
```diff
@@ -23,7 +24,7 @@ export const useApiKeyConfig = () => {
   const { publish } = useToast();
   const [userIdentity, setUserIdentity] = useState<UserIdentity>();
   const { t } = useTranslation(["config"]);
-
+  const gaEventTracker = useGAnalyticsEventTracker("QUIVR_API_KEY");
```
Comment (StanGirard) [inline frontend/app/user/components/ApiKeyConfig/hooks/useApiKeyConfig.ts:27]: @gozineb what is that ?

Comment (StanGirard) [reply frontend/app/user/components/ApiKeyConfig/hooks/useApiKeyConfig.ts:27]: It is not declared

**Review by StanGirard — CHANGES_REQUESTED** [review]

File: frontend/app/user/components/ApiKeyConfig/hooks/useApiKeyConfig.ts [hunk 2]
```diff
@@ -30,6 +31,7 @@ export const useApiKeyConfig = () => {
     queryKey: [USER_IDENTITY_DATA_KEY],
     queryFn: getUserIdentity,
   });
+  const gaEventTracker = useGAnalyticsEventTracker("QUIVR_API_KEY");
```
Comment (mamadoudicko) [inline frontend/app/user/components/ApiKeyConfig/hooks/useApiKeyConfig.ts:34]: Maybe 
```ts
  const {trackEvent } = useGAnalyticsEventTracker({
    eventsCategory: "QUIVR_API_KEY",
  })
```

File: frontend/services/analytics/google/googleAnalytics.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          window.dataLayer = window.dataLayer || [];
+          function gtag(){dataLayer.push(arguments);}
+          gtag('js', new Date());
+          gtag('config', '${ga_id}');
+        `,
+        }}
+      />
+    </>
+  );
+};
```
Comment (mamadoudicko) [inline frontend/services/analytics/google/googleAnalytics.tsx:29]: Seems to be a duplication. Already done in `ReactGA.initialize(ga_id)` in `useGAnalyticsEventTracker`

**Comment by mamadoudicko** [comment]
Some frontend tests need to be updated: Vitest step is falling

**Review by mamadoudicko — DISMISSED** [review]

**Review by StanGirard — APPROVED** [review]

**Pull request merged**